test/screenshot/screenshot.js

```diff
@@ -1,60 +1,220 @@
+    goldenJSON[this.urlPath_] = goldenId;
+    await writeFilePromise(GOLDENS_FILE_PATH, JSON.stringify(goldenJSON, null, '  '));
+  }
+
+  /**
+   * Takes a screenshot of the URL
+   * @return {!Buffer}
+   * @private
+   */
+  async createScreenshotTask_() {
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
+    await page.goto(`http://localhost:8080/${this.urlPath_}`);
+    const imageBuffer = await page.screenshot();
+    await browser.close();
+    return imageBuffer;
+  }
+
+  /**
+   * Generates a unique hash from an image's contents
+   * @param {!Buffer} imageBuffer The image buffer to hash
+   * @return {string}
+   * @private
+   */
+  generateImageID_(imageBuffer) {
+    return createHash('sha256').update(imageBuffer).digest('hex');
+  }
+
+  /**
+   * Returns the correct image path
+   * @param {string} imageId The image ID
+   * @param {string} imageType The image type
+   * @private
+   */
+  getImagePath_(imageId, imageType) {
+    switch (imageType) {
+      case GOLDEN:
+      return `${this.urlPath_}/${imageId}.golden.png`;
+
+      case SNAPSHOT:
+      case DIFF:
+      return `${this.urlPath_}/${COMMIT_HASH}/${imageId}.${imageType}.png`;
+    }
+  }
+
+  /**
+   * Saves the given image to Google Cloud Storage with optional metadata
+   * @param {string} imagePath The path to the image
+   * @param {!Buffer} imageBuffer The image buffer
+   * @param {Object} customMetadata Optional custom metadata
+   * @private
+   */
+  async saveImage_(imagePath, imageBuffer, customMetadata={}) {
+    const metadata = Object.assign({}, DEFAULT_METADATA, customMetadata);
+    const file = bucket.file(imagePath);
+
+    // Check if file exists and exit if it does
+    const [exists] = await file.exists();
+    if (exists) {
+      console.log('✔︎ No changes', imagePath);
+      return;
+    }
+
+    // Create a new stream from the image buffer
+    let stream = new Readable();
+    stream.push(imageBuffer);
+    stream.push(null);
+
+    // The promise is resolved or rejected inside the stream event callbacks
+    return new Promise((resolve, reject) => {
+      stream.pipe(file.createWriteStream())
+        .on('error', (err) => {
+          reject(err);
+        }).on('finish', async () => {
+          try {
+            // Make the image public and set it's metadata
+            await file.makePublic();
+            await file.setMetadata({metadata});
+            console.log('✔︎ Uploaded', imagePath);
+            resolve();
+          } catch (err) {
+            reject(err);
+          }
+        });
+    });
+  }
+
+  /**
+   * Downloads an image from Google Cloud Storage
+   * @param {string} gcsFilePath The file path on Google Cloud Storage
+   * @return {Buffer}
+   * @private
+   */
+  async readImage_(gcsFilePath) {
+    const data = await bucket.file(gcsFilePath).download();
+    return data[0];
+  }
+
+  /**
+   * Captures a screenshot of the test URL
+   */
   capture() {
```

Comment:
The name `capture` is misleading. It would be clearer to name it something like `updateGolden`.